manifests/params.pp

```diff
@@ -64,16 +73,35 @@
     }
 
     default : {
-      $dir        = '/etc/puppet'
-      $logdir     = '/var/log/puppet'
-      $rundir     = '/var/run/puppet'
-      $ssldir     = '/var/lib/puppet/ssl'
-      $vardir     = '/var/lib/puppet'
-      $sharedir   = '/usr/share/puppet'
+      if $aio_package {
+        $dir      = '/etc/puppetlabs/puppet'
+        $codedir  = '/etc/puppetlabs/code'
+        $logdir   = '/var/log/puppetlabs/puppet'
+        $rundir   = '/var/run/puppetlabs'
+        $ssldir   = '/etc/puppetlabs/puppet/ssl'
+        $vardir   = '/opt/puppetlabs/puppet/cache'
+        $sharedir = '/opt/puppetlabs/puppet'
+      } else {
+        $dir        = '/etc/puppet'
+        $codedir    = '/etc/puppet'
+        $logdir     = '/var/log/puppet'
+        $rundir     = '/var/run/puppet'
+        $ssldir     = '/var/lib/puppet/ssl'
+        $vardir     = '/var/lib/puppet'
+        $sharedir   = '/usr/share/puppet'
+      }
       $root_group = undef
     }
   }
 
+  if $aio_package and $::osfamily != 'Windows' {
+    $dir_owner = 'root'
+    $dir_group = $root_group
+  } else {
+    $dir_owner = $user
+    $dir_group = $group
+  }
```

Comment:
On Windows we do not want to set the owner explicitly and there is no user named puppet so, as it stands, this else block causes a failed run. If you make the following change it fixes it:

```puppet
if $aio_package and $::osfamily != 'Windows' {
  $dir_owner = 'root'
  $dir_group = $root_group
} else {
  $dir_owner = undef
  $dir_group = undef
}
```